content/main/docs/serverless-workflows/configuration/token-propagation.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Configure workflow for token propagation"
+date: 2025-04-04
+---
+The RHDH Orchestrator plugin is, by default, adding all tokens in the `authTokens` field of the POST request's body in the workflow execuition's request with the following format: `X-Authorization-{provider}: {token}`.
+This allows the user identity to be propagated to the third parties/externals services called by the workflow.
+To do so, a set of properties must be set in the workflow `application.properties` file.
+
+# Prerequisites
+* Having a Keycloak instance running with a client
+* Having RHDH with the latest version of the Orchestrator plugins
+* Having a workflow using openapi spec file to send REST requests to a service
+
+# Build
+
+When building the workflow's image, you will need to make sure the following extensions is present in the `QUARKUS_EXTENSION`:
+* io.quarkus:quarkus-oidc-client-filter # needed for propagation
+* io.quarkus:quarkus-oidc # neded for token validity check thus accessing $WORKFLOW.identity
+
+# Configuration
+## Oauth2
+1. In the openapi spec file(s) for which you want to propagate the incoming token to, write down the security scheme configured for the endpoints of interests for you. All endpoints may use the same security scheme if configured globally.
+e.g
+```
+components:
+  securitySchemes:
+    BearerToken:
+     type: oauth2
+     flows:
+       clientCredentials:
+         tokenUrl: http://<keycloak>/realms/<yourRealm>/protocol/openid-connect/token
+         scopes: {}
+     description: Bearer Token authentication
+```
+2. In the `application.properties` of your workflow, for each security scheme, adds the following:
```

Comment:
are there any sonataFlow docs that we can refer to?

Reply:
Yes, added